pkg/prow/apiv1/branch_protection.go

```diff
@@ -0,0 +1,307 @@
+package apiv1
```

Comment:
Put all of the updates to this package in their own commit, describe where it comes from `prowapiv1: Updated from ci-chat-bot` and include the commit from ci-chat-bot in the description.